x/stakeibc/keeper/icqcallbacks_delegator_shares.go

```diff
@@ -26,108 +25,99 @@ import (
-	} else if !isWithinWindow {
-		k.Logger(ctx).Error("delegator shares callback is outside ICQ window")
-		return nil
+		return sdkerrors.Wrapf(sdkerrors.ErrInvalidRequest, "unable to determine if ICQ callback is inside buffer window, err: %s", err.Error())
+	}
+	if !withinBufferWindow {
+		// QUESTION: Should this return an error?
+		return sdkerrors.Wrapf(types.ErrOutsideIcqWindow, "callback is outside ICQ window")
 	}
 
-	// Grab the validator object form the hostZone using the address returned from the query
+	// Grab the validator object from the hostZone using the address returned from the query
 	validator, valIndex, found := GetValidatorFromAddress(hostZone.Validators, queriedDelgation.ValidatorAddress)
 	if !found {
-		errMsg := fmt.Sprintf("no registered validator for address (%s)", queriedDelgation.ValidatorAddress)
-		k.Logger(ctx).Error(errMsg)
-		return sdkerrors.Wrapf(types.ErrValidatorNotFound, errMsg)
+		return sdkerrors.Wrapf(types.ErrValidatorNotFound, "no registered validator for address (%s)", queriedDelgation.ValidatorAddress)
 	}
 
-	// get the validator's internal exchange rate, aborting if it hasn't been updated this epoch
+	// Get the validator's internal exchange rate, aborting if it hasn't been updated this epoch
 	strideEpochTracker, found := k.GetEpochTracker(ctx, epochtypes.STRIDE_EPOCH)
 	if !found {
-		k.Logger(ctx).Error("failed to find stride epoch")
-		return sdkerrors.Wrapf(sdkerrors.ErrNotFound, "no epoch number for epoch (%s)", epochtypes.STRIDE_EPOCH)
+		return sdkerrors.Wrapf(sdkerrors.ErrNotFound, "unable to get epoch tracker for epoch (%s)", epochtypes.STRIDE_EPOCH)
 	}
-	if validator.InternalExchangeRate.EpochNumber != strideEpochTracker.GetEpochNumber() {
-		errMsg := fmt.Sprintf("DelegationCallback: validator (%s) internal exchange rate has not been updated this epoch (epoch #%d)",
-			validator.Address, strideEpochTracker.GetEpochNumber())
-		k.Logger(ctx).Error(errMsg)
-		return sdkerrors.Wrapf(sdkerrors.ErrInvalidRequest, errMsg)
+	if validator.InternalExchangeRate.EpochNumber != strideEpochTracker.EpochNumber {
+		return sdkerrors.Wrapf(sdkerrors.ErrInvalidRequest,
+			"validator (%s) internal exchange rate has not been updated this epoch (epoch #%d)", validator.Address, strideEpochTracker.EpochNumber)
 	}
 
-	// TODO: make sure conversion math precision matches the sdk's staking module's version (we did it slightly differently)
-	// note: truncateInt per https://github.com/cosmos/cosmos-sdk/blob/cb31043d35bad90c4daa923bb109f38fd092feda/x/staking/types/validator.go#L431
-	validatorTokens := queriedDelgation.Shares.Mul(validator.InternalExchangeRate.InternalTokensToSharesRate).TruncateInt()
-	k.Logger(ctx).Info(fmt.Sprintf("DelegationCallback: HostZone: %s, Validator: %s, Previous NumTokens: %v, Current NumTokens: %v",
-		hostZone.ChainId, validator.Address, validator.DelegationAmt, validatorTokens))
+	// Calculate the number of tokens delegated (using the internal exchange rate)
```

Comment:
Slight preference to include the link / reference to the SDK function we're replicating, since `tokensToShares` is a bit of a confusing concept

Reply:
Ah good call, did not mean to remove that, just meant to remove the TODO.

Adding back in

Reply:
Thanks!